### DISCUSSION TITLE
Installing Security Onion 2 on Physical Host

### DISCUSSION THREAD

**Discussion by veraxcyber** [Security-Onion-Solutions/securityonion#4127] [Unsupported Versions] [answered]
Hello, 

When I try to install Security Onion 2 on a physical host, the ISO doesn't recognise UEFI as a boot option and will only boot in legacy mode. 

I used Etcher and Rufus to create the bootable media but I am getting the "Minimal BASH like line editing is supported GRUB" error. 

Has anyone setup security onion 2 on a physical host and if so how did you achieve this please?

Many Thanks

**Comment by jtournade** [comment]
Did you find the solution? I'm having the same problem.

**Comment by veraxcyber** [comment]
Hey,

Yeah you will need to change the BIOS to legacy and then boot from the USB.

Thanks!

**Comment by gorbehnare** [comment]
Hello

After hours of fiddling with this today, I also discovered that this does not boot in UEFI. This is bad, because my new server is UEFI only. It looks like the only solution is to do a manual install (install Ubuntu or Centos, then install software on top). I don't have time for this since this is just for testing right now, but I had to dig up an old OptiPlex desktop to run this... I'm not happy, and we will probably not use this solution. 

What bad first impression...

**Comment by dougburks** [comment, marked as answer]
@gorbehnare Security Onion 2 has supported UEFI since version 2.3.0:
https://docs.securityonion.net/en/2.3/release-notes.html#id354

So assuming you're using the latest version of Security Onion 2, it's most likely not a UEFI issue and more likely some other aspect of your hardware.

If you have further questions or problems, please review https://github.com/Security-Onion-Solutions/securityonion/discussions/1720 and then start a new discussion rather than replying to an old discussion.

**Reply by kstryfe** [reply]
I just attempted to boot the ISO (verified) for the latest version (2.4.50-20240220) off of USB (using Balena Etcher per instructions). It fails to boot in UEFI on multiple machines  (2 physical of different brand as well as 2 virtual machines of different hypervisors), but boots properly in legacy mode. 

The device shows as a boot option, but then behaves as though it is missing the actual bootcode. The BIOS simply skips to the next device, or if given no other device, complains about having no bootable device. 

I have tried multiple flashdrives and have also used alternate methods to produce a bootable USB from the ISO (dd, rufus) with no better results. The virtual machines were attempted to be booted from the ISO directly as well as USB passthrough. 

This points to a misconfiguration of the ISO for UEFI boot. If I have the time, I might try to dig a little and see if I can offer a more specific fault, but it has been busy of late.

**Reply by dougburks** [reply]
I've just tested the 2.4.50-20240220 ISO image and can verify that it boots correctly on my UEFI machine.

Secure Boot disabled:

![image](https://github.com/Security-Onion-Solutions/securityonion/assets/1659467/2655914c-0772-4eab-837c-68f196f47d9a)

Secure Boot enabled:

![image](https://github.com/Security-Onion-Solutions/securityonion/assets/1659467/2eb00da4-ff24-491d-aec7-0b4df9b03255)

If you have further questions or problems, please start a new discussion rather than replying to this old discussion from 2021/2022.
